cmd/snap/inhibit.go

```diff
@@ -82,6 +75,12 @@ func waitWhileInhibited(ctx context.Context, cli *client.Client, snapName string
 		return err
 	}
 	inhibited := func(ctx context.Context, hint runinhibit.Hint, inhibitInfo *runinhibit.InhibitInfo) (cont bool, err error) {
+		if hint == runinhibit.HintInhibitedForRemove {
+			return false, errInhibitedForRemove
+		}
+		if !features.RefreshAppAwareness.IsEnabled() {
+			return true, errOngoingSnapRefresh
+		}
```

Comment:
If the snap is inhibited but RAA is not enabled, then where are `info` and `app` set? Previously, they were set in `maybeWaitWhileInhibited`, but not they're only set if the `notInhibited` callback is called.

AFAICT if RAA is not enabled and the snap is inhibited then `runinhibitWaitWhileInhibited` calls `inhibited` is called, and it immediately returns with a `nil` error. But since `notified` can only be true if `inhibited` was called with RAA enabled, then there's no way to get to that error case, so the only return value is `info` and `app`, which have not been set.

Reply:
if RAA is not enabled, an error is returned in `inhibited` callback (`errOngoingSnapRefresh`) and `snap run` would refuse to start.

Previously it was setting `info` and `app` which would sometimes fail, and sometimes not, this depended on how deep the refresh went (the current symlink is removed after `unlink-current-snap` task). I changed the logic slightly, that it would always fail to start if the snap is inhibited.

Reply:
You make a really good point, that we never check if `info` and `app` are non-nil before returning, would it make sense to check and return a sensible `internal error` message?

Reply:
> Previously it was setting info and app which would sometimes fail, and sometimes not

Actually it would always fail, the current symlink is removed just after the snap is inhibited